### PR TITLE
Ab 625 app status new UI

### PR DESCRIPTION
**Pull request by vivek-devtron**
# Description
App details new UI with filters and sorting. Use same for all the app detail page

Fixes # [(AB#625)](https://dev.azure.com/DevtronLabs/Devtron/_workitems/edit/625)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
manually tested all the scenarios

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/details/appDetails/DeploymentStatusDetailModal.tsx [hunk 1]
```diff
@@ -13,9 +13,39 @@ export default function DeploymentStatusDetailModal({
...
+        document.addEventListener('click', outsideClickHandler)
+        return (): void => {
+            document.removeEventListener('click', outsideClickHandler)
+        }
+    }, [outsideClickHandler])
+
     return (
-        <Drawer position="right" width="50%">
-            <div className="deployment-status-breakdown-modal-container bcn-0">
+        <Drawer position="right" width="1024px">
```
Comment (shivani170) [inline src/components/app/details/appDetails/DeploymentStatusDetailModal.tsx:47]: Hope this strictly providing width will work fine in all screens

File: src/components/v2/appDetails/k8Resource/StatusFilterButton.component.tsx [hunk 2]
```diff
@@ -1,51 +1,61 @@
...
+    ]
 
     // const handleFilterClick = (filterName: string) => {
     //     IndexStore.updateFilterType(filterName);
     //     setSelectedTab(filterName.toLowerCase());
     // };
 
-    useEffect(()=>{
+    useEffect(() => {
         // handleFilterClick(selectedTab.toUpperCase())
```
Comment (shivani170) [inline src/components/v2/appDetails/k8Resource/StatusFilterButton.component.tsx:52]: if not in use we can remove this commented code

File: src/components/v2/appDetails/k8Resource/StatusFilterButton.component.tsx [hunk 3]
```diff
@@ -59,9 +69,9 @@ export const StatusFilterButtonComponent = ({nodes}: {nodes: Array<Node>}) => {
                                         setSelectedTab(filter.status.toLowerCase())
                                         // handleFilterClick(filter.status);
```
Comment (shivani170) [inline src/components/v2/appDetails/k8Resource/StatusFilterButton.component.tsx:70]: Here as well

File: src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx [hunk 4]
```diff
@@ -18,12 +20,44 @@ interface NodeStreamMap {
...
     const _appDetails = IndexStore.getAppDetails()
 
     const nodes: AggregatedNodes = useMemo(() => {
         return aggregateNodes(_appDetails.resourceTree?.nodes || [], _appDetails.resourceTree?.podMetadata || [])
     }, [_appDetails])
+    const nodesKeyArray = Object.keys(nodes?.nodes || {})
+    let flattenedNodes = []
+    if (nodesKeyArray.length > 0) {
+        for (let index = 0; index < nodesKeyArray.length; index++) {
+            //if(nodesKeyArray[index].toLowerCase() !== 'rollout'){
```
Comment (shivani170) [inline src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx:47]: Please remove commented code

File: src/components/app/details/appDetails/DeploymentStatusDetailModal.tsx (hunk 1 above)
Comment (shivani170) [inline src/components/app/details/appDetails/DeploymentStatusDetailModal.tsx:47]: Is hard coded width working fine in all screens?

File: src/components/v2/appDetails/k8Resource/StatusFilterButton.component.tsx [hunk 5]
```diff
@@ -1,51 +1,61 @@
...
 
-export const StatusFilterButtonComponent = ({nodes}: {nodes: Array<Node>}) => {
-    const [selectedTab, setSelectedTab] = useState('all');
+export const StatusFilterButtonComponent = ({
+    nodes,
+    handleFilterClick,
+}: {
+    nodes: Array<Node>
+    handleFilterClick?: (selectedFilter: string) => void
+}) => {
```
Comment (sohel-rp) [inline src/components/v2/appDetails/k8Resource/StatusFilterButton.component.tsx:18]: move to types

File: src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx [hunk 6]
```diff
@@ -18,12 +20,44 @@ interface NodeStreamMap {
...
+
+function AppStatusDetailModal({
+    close,
+    appStreamData,
+    showAppStatusMessage,
+}: {
+    close: () => void
+    appStreamData: any
+    showAppStatusMessage?: boolean
+}) {
```
Comment (sohel-rp) [inline src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx:37]: move to types

File: src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx [hunk 7]
```diff
@@ -18,12 +20,44 @@ interface NodeStreamMap {
...
     const _appDetails = IndexStore.getAppDetails()
 
     const nodes: AggregatedNodes = useMemo(() => {
         return aggregateNodes(_appDetails.resourceTree?.nodes || [], _appDetails.resourceTree?.podMetadata || [])
     }, [_appDetails])
+    const nodesKeyArray = Object.keys(nodes?.nodes || {})
```
Comment (sohel-rp) [inline src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx:43]: can we change this to something like - `nodes?.nodes ? Object.keys(nodes.nodes) : []`?

File: src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx (hunk 4 above)
Comment (sohel-rp) [inline src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx:47]: remove commented

File: src/components/app/details/appDetails/DeploymentStatusDetailModal.tsx (hunk 1 above)
Comment (vivek-devtron) [reply src/components/app/details/appDetails/DeploymentStatusDetailModal.tsx:47]: yes

Comment (vivek-devtron) [reply src/components/app/details/appDetails/DeploymentStatusDetailModal.tsx:47]: It is as per my discussions with Utkarsh. in next iteration may be we will be adding some minmax funda

File: src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx [hunk 8]
```diff
@@ -95,6 +129,21 @@ function AppStatusDetailModal({ close, appStreamData, showAppStatusMessage }: {
...
 
+    const onFilterClick = (selectedFilter: string): void => {
+        if (currentFilter !== selectedFilter) {
+            setCurrentFilter(selectedFilter)
+            if (selectedFilter === 'ALL') {
+                setFilteredNodes(flattenedNodes)
+            } else {
+                setFilteredNodes(
+                    flattenedNodes.filter(
+                        (nodeDetails) => nodeDetails.health?.status?.toLowerCase() === selectedFilter.toLowerCase(),
```
Comment (sohel-rp) [inline src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx:140]: Do we need an optional check here? If yes the please please add the same while sorting at line 54.

Comment (vivek-devtron) [reply src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx:135]: Done

File: src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx (hunk 4 above)
Comment (vivek-devtron) [reply src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx]: Done

File: src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx (hunk 6 above)
Comment (vivek-devtron) [reply src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx]: Done

File: src/components/v2/appDetails/k8Resource/StatusFilterButton.component.tsx (hunk 5 above)
Comment (vivek-devtron) [reply src/components/v2/appDetails/k8Resource/StatusFilterButton.component.tsx]: Done

File: src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx (hunk 4 above)
Comment (vivek-devtron) [reply src/components/v2/appDetails/sourceInfo/environmentStatus/AppStatusDetailModal.tsx]: Done

File: src/components/v2/appDetails/k8Resource/StatusFilterButton.component.tsx (hunk 3 above)
Comment (vivek-devtron) [reply src/components/v2/appDetails/k8Resource/StatusFilterButton.component.tsx]: Done

File: src/components/v2/appDetails/k8Resource/StatusFilterButton.component.tsx (hunk 2 above)
Comment (vivek-devtron) [reply src/components/v2/appDetails/k8Resource/StatusFilterButton.component.tsx]: Done

**Review by sohel-rp — APPROVED** [review]

**Review by shivani170 — APPROVED** [review]

**Pull request merged**